api_management/apps/analytics/filters.py

```diff
@@ -0,0 +1,78 @@
+import datetime
+from abc import abstractmethod
+
+from django.core.exceptions import ImproperlyConfigured
+from rest_framework.filters import BaseFilterBackend
+
+
+class QueryParamBaseFilterBackend(BaseFilterBackend):
```

Comment:
[django_filter](https://django-filter.readthedocs.io/en/master/guide/usage.html#declaring-filters) nos permite evitar la implementacion de estas clases y los checkeos de `if params is None`. Las combinaciones entre field y un lookup hacen claara y sensilla la implementacion. Fijate si podes pasarlo a eso.